packages/mds-cache/index.ts

```diff
@@ -175,33 +189,57 @@ async function writeDeviceState(field: UUID, data: StateEntry) {
   return (await getClient()).hsetAsync(decorateKey('device:state'), field, JSON.stringify(data))
 }
 
-async function readTripsEvents(field: UUID): Promise<TripsEvents | null> {
-  const tripsEvents = await hget(decorateKey('trips:events'), field)
-  return tripsEvents ? parseTripsEvents(tripsEvents as StringifiedTripsEvents) : null
+async function readTripEvents(field: UUID): Promise<TripEvent[] | null> {
+  const tripEvents = await hget(decorateKey('trips:events'), field)
+  return tripEvents ? parseTripEvents(tripEvents as StringifiedTripEvents) : null
 }
 
-async function readAllTripsEvents(): Promise<{ [vehicle_id: string]: TripsEvents } | null> {
+// TODO: Investigate alternatives, this should be used temporarily. O(n) runtime for streaming is not ideal.
+async function readDeviceTripsEvents(pattern: string): Promise<TripsEvents | null> {
+  // hscan returns list of alternating key and value strings (i.e. [keyA, valueA, keyB, valueB])
+  const allFilteredTripEventsList = await hscan('trips:events', pattern)
+  if (allFilteredTripEventsList) {
```

Comment:
I kinda feel like using `reduce` here is unnecessarily clever.  I could be wrong.  A dopey `for` loop would be less code.